jax/_src/compiler.py

```diff
@@ -53,6 +53,17 @@
     ),
 )
 
+_ENABLE_COMPILER_REMAT_OPTIMIZATION_PASS = config.bool_flag(
+    "jax_compiler_enable_remat_pass",
+    config.bool_env('JAX_COMPILER_ENABLE_REMAT_PASS', False),
```

Comment:
default should be True, because that's the situation at HEAD IIUC. You can switch it to False for your use case.

Reply:
There was discussion with @hawkinsp that we want to change the default in JAX.
The current default is bad for a few reasons.
It isn't predictable as small change to the graph give wrong reasons.
Currently, we always disable it as it trigger too easily and does a poor selection of what to remat.

So the conclusion was to not enable it by default and document that users can enable it if they want to try it.